### PR TITLE
Updated Username Validation Logic

### PR DESCRIPTION
**Pull request by victor-oancea-leftclick-io**
transformed PUT /v1/user-validations/username endpoint into GET /v1/user-views/username?username=something that validates the username and return some public information for the user that has that username

### CODE REVIEW

File: cmd/eskimo/contract.go [hunk 1]
```diff
@@ -3,17 +3,23 @@
...
 
 // Public API.
 
 type (
-	RequestGetUser struct {
+	RequestGetUserByID struct {
 		AuthenticatedUser server.AuthenticatedUser `json:"authenticatedUser" swaggerignore:"true"`
 		ID                string                   `uri:"userId" example:"did:ethr:0x4B73C58370AEfcEf86A6021afCDe5673511376B2"`
 	}
+	RequestGetByUsername struct {
```
Comment (azayats-leftclick) [inline cmd/eskimo/contract.go:19]: RequestGetUserByUsername?

Comment (victor-oancea-leftclick-io) [reply cmd/eskimo/contract.go:19]: right

File: cmd/eskimo/contract.go [hunk 2]
```diff
@@ -36,11 +42,14 @@ type (
 const (
 	applicationYamlKey = "cmd/eskimo"
 	userNotFoundCode   = "USER_NOT_FOUND"
+	usernameRegex      = `^[\w\-.]{4,20}$`
```
Comment (azayats-leftclick) [inline cmd/eskimo/contract.go:45]: "...." - is it a valid username?

Comment (victor-oancea-leftclick-io) [reply cmd/eskimo/contract.go:45]: we'll improve the regex later on

File: cmd/eskimo/users.go [hunk 3]
```diff
@@ -69,24 +70,96 @@ func (s *service) GetUser(ctx context.Context, r server.ParsedRequest) server.Re
...
+	}
+}
+
+func (req *RequestGetByUsername) GetAuthenticatedUser() server.AuthenticatedUser {
+	return req.AuthenticatedUser
+}
+
+func (req *RequestGetByUsername) Validate() *server.Response {
+	if !compiledUsernameRegex.MatchString(req.Username) {
+		err := errors.Errorf("username: %v is invalid, it should match regex: %v", req.Username, usernameRegex)
```
Comment (azayats-leftclick) [inline cmd/eskimo/users.go:149]: Can the user understand the regexp in the error message?

File: users/users_get.go [hunk 4]
```diff
@@ -26,17 +27,23 @@ func (u *users) GetUser(ctx context.Context, id UserID) (*User, error) {
...
+func (u *users) GetUserByUsername(ctx context.Context, name Username) (*User, error) {
 	if ctx.Err() != nil {
-		return false, errors.Wrap(ctx.Err(), "get user failed because context failed")
+		return nil, errors.Wrap(ctx.Err(), "get user failed because context failed")
 	}
 	result := new(user)
 
-	if err := u.db.GetTyped("USERS", "USERS_USERNAME_LOOKUP_IX", []interface{}{name}, result); err != nil {
-		return false, errors.Wrapf(err, "failed to get user")
+	if err := u.db.GetTyped("USERS", "unique_unnamed_USERS_2", tarantool.StringKey{S: name}, result); err != nil {
```
Comment (azayats-leftclick) [inline users/users_get.go:36]: no such index in DDL
will create it later on?

Comment (victor-oancea-leftclick-io) [reply users/users_get.go:36]: it is there, tarantool automatically creates indexes for primary keys and unique keys, that last number is the index counter from top to bottom of the create table ddl, starting with 1

Comment (azayats-leftclick) [reply users/users_get.go:36]: I thought the same, but get this with my local try
http://joxi.ru/EA4lqlaTX9e1KA

Comment (victor-oancea-leftclick-io) [reply users/users_get.go:36]: use the cli to get all the details, sql driver is limited


you can connect to a running process, check the readme

Comment (victor-oancea-leftclick-io) [reply users/users_get.go:36]:
```
docker exec -t -i mytarantool console
``` 

where mytarantool is the container name

Comment (victor-oancea-leftclick-io) [reply users/users_get.go:36]: then run ```

box.space.USERS

```from inside

Comment (azayats-leftclick) [reply users/users_get.go:36]: got it, thanks!

File: cmd/eskimo/users.go [hunk 5]
```diff
@@ -17,10 +17,11 @@ import (
 func (s *service) setupUserRoutes(router *gin.Engine) {
 	router.
 		Group("/v1").
-		GET("users/:userId", server.RootHandler(newRequestGetUser, s.GetUser))
+		GET("users/:userId", server.RootHandler(newRequestGetUserByID, s.GetUserByID)).
+		GET("user-views/username", server.RootHandler(newRequestGetByUsername, s.GetUserByUsername))
```
Comment (azayats-leftclick) [inline cmd/eskimo/users.go:21]: - GET("users/username", ...)
- GET("user/:userName", ...)
?

Comment (victor-oancea-leftclick-io) [reply cmd/eskimo/users.go:21]: neither are RESTful

File: users/users_get.go (hunk 4 above)
Comment (kfedor) [inline users/users_get.go:36]: is this some default index we don't have to create?

Comment (azayats-leftclick) [reply users/users_get.go:36]: Yeah, we've discussed this with Victor.
Take a look at resolved conversation above.

**Review by kfedor — APPROVED** [review]

File: cmd/eskimo/users.go (hunk 3 above)
Comment (victor-oancea-leftclick-io) [reply cmd/eskimo/users.go:149]: He’d better, the user’s a senior front end developer

**Review by azayats-leftclick — APPROVED** [review]

**Review by kfedor — APPROVED** [review]

**Pull request merged**